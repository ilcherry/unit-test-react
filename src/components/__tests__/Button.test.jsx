// Button.test.js
import { render } from "@testing-library/react";
import Button from "../Button/Button";

describe("Button Component Snapshot", () => {
  it("should match the snapshot", () => {
    // 渲染组件
    const { asFragment } = render(<Button label="Click Me" />);

    // 使用快照 API 将渲染结果与之前保存的快照进行比较
    expect(asFragment()).toMatchSnapshot();
  });
});
