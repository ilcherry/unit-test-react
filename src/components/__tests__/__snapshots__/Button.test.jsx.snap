// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component Snapshot > should match the snapshot 1`] = `
<DocumentFragment>
  <button>
    Click Me
  </button>
</DocumentFragment>
`;
